### PR TITLE
Guard against null array

### PR DESCRIPTION
**Pull request by pinzart90**
### Purpose

https://jira.autodesk.com/browse/DYN-4475
Guard against null array

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoManipulation/MousePointManipulator.cs [hunk 1]
```diff
@@ -23,7 +23,8 @@ public class MousePointManipulator : NodeManipulator
 
         private TranslationGizmo gizmo;
 
-        private Dictionary<int, Tuple<Vector, NodeModel>> indexedAxisNodePairs;
+        //Holds manipulator axis and input node pair for each input port.
+        private Dictionary<int, Tuple<Vector, NodeModel>> indexedAxisNodePairs = new Dictionary<int, Tuple<Vector, NodeModel>>();
```
Comment (pinzart90) [inline src/DynamoManipulation/MousePointManipulator.cs:27]: we will always have an empty indexedAxisNodePairs Dictionary

File: src/DynamoManipulation/MousePointManipulator.cs [hunk 2]
```diff
@@ -36,8 +37,8 @@ protected override void AssignInputNodes()
         {
             //Default axes
             var axes = new Vector[] { Vector.XAxis(), Vector.YAxis(), Vector.ZAxis() };
-            //Holds manipulator axis and input node pair for each input port.
-            indexedAxisNodePairs = new Dictionary<int, Tuple<Vector, NodeModel>>(3);
+
+            indexedAxisNodePairs.Clear();
```
Comment (pinzart90) [inline src/DynamoManipulation/MousePointManipulator.cs:41]: just need to clear it where it was previously being recreated

Comment (aparajit-pratap) [reply src/DynamoManipulation/MousePointManipulator.cs:41]: How does this fix the crash, not sure I understand?

Comment (pinzart90) [reply src/DynamoManipulation/MousePointManipulator.cs:41]: indexedAxisNodePairs will now be created when the constructor of the class is called.
So indexedAxisNodePairs will not be null anymore, only empty (which is properly handled)

**Review by aparajit-pratap — APPROVED** [review]

**Pull request merged**